### ISSUE TITLE
implement useRoots or something like that

### ISSUE THREAD

**Issue by ged-odoo** [odoo/owl#1260]
```js
setup() {
  this.roots = useRoots(); // now: { elems: [] }

  onMounted(() => {
      console.log(this.roots); // {elems: [el]}, or {elems: [el1, el2, ...,eln]} if multiroot
      // can be used like this:
      const [el] = this.roots.elems;

  });
}

**Comment by ged-odoo** [comment]
the result of this.roots would be an object with a getter, so the value of elems is computed when we read it, and is always uptodate

**Comment by ged-odoo** [comment]
main goal is to make it easier to use higher order components that need to manipulate the el for their subcomponent, for example add classes on the concrete component in `Field`

**Comment by sdegueldre** [comment]
Probably should return nodes. You've basically already implemented all the tech stuff for this within event catcher right?

**Comment by ged-odoo** [comment]
exactly, i was thinking about extracting the "range of elements" concept of event catcher, so it can be reused here.
now, it could return nodes, but the main usecase would certainly be htmlelements.  We could also return both, since it would be a getter:
```js
  {
    elems: [...],
    nodes: [...],
  }
```
